### PR TITLE
MRG: Factor out eeg_loc and coil_trans

### PR DESCRIPTION
**Pull request by Eric89GXL**
The `ch['eeg_loc']` and `ch['coil_trans']` values are redundant with `ch['loc']`, so keeping them around is dangerous. It simplifies the code to only keep `ch['loc']` around, which is what this PR does.

### CODE REVIEW

File: mne/realtime/fieldtrip_client.py [hunk 1]
```diff
@@ -1,353 +1,351 @@
-# Author: Mainak Jas
-#
-# License: BSD (3-clause)
```
Comment (Eric89GXL) [inline mne/realtime/fieldtrip_client.py]: This file was auto-changed by Spyder because it had mixed line endings. I can revert if necessary.

Comment (agramfort) [inline mne/realtime/fieldtrip_client.py]: yes please revert.

**Comment by agramfort** [comment]
travis is not happy. Did not look why

**Comment by Eric89GXL** [comment]
@agramfort Travis is all better. Happy otherwise?

@dengemann you should probably also take a look since it touches your BTI code.

**Comment by dengemann** [comment]
@Eric89GXL let me get my open PR in first please ... would be easier I feel.

**Comment by Eric89GXL** [comment]
Do you have time to work on that soon? This one is ready to go...

**Comment by dengemann** [comment]
> Do you have time to work on that soon? This one is ready to go...

On wednesday. If you feel it's urgent move on and I'll bite the bullet of rebasing later.

**Comment by Eric89GXL** [comment]
I could also do the rebase for you if you want

**Comment by dengemann** [comment]
> I could also do the rebase for you if you want

Feel free to move on, on the bti branch I'm conceptually done. You've got push access, feel at home.

**Comment by Eric89GXL** [comment]
Cool, I'll give it a shot. I'm sure the tests you wrote are complete enough that if I break anything it will be obvious right? :)

**Comment by dengemann** [comment]
Yeah Eric, the tests were good enough I'd say.

**Comment by wmvanvliet** [comment]
anything that simplifies the info structure is a good thing™

**Comment by agramfort** [comment]
does it make any difference for the files written to disk?

@mshamalainen any reason why these values are redundant?

**Comment by Eric89GXL** [comment]
@agramfort no, only `loc` ever gets written out. I think it's just convenient to have `coil_trans` and `eeg_loc` accessible/parsed from `loc`, but with the modifications being made to `coil_trans` it in `bti` and now `maxfilter` it's safer to convert to trans (really just reorder) and then convert back to `loc` (again a reorder).

**Comment by agramfort** [comment]
Ok

**Comment by Eric89GXL** [comment]
Rebased and tests pass over here, hopefully the CIs agree. @agramfort @dengemann feel free to have another look when you get a chance.

**Comment by agramfort** [comment]
have a look at this:

https://github.com/mne-tools/mne-python/blob/master/doc/manual/matlab.rst

it documents loc, eeg_loc and coil_trans. Should we be concerned about the
differences ?

I am a bit clueless about the origin of this.... ping @mshamalainen

**Comment by Eric89GXL** [comment]
`eeg_loc` is the first six values of `loc` reshaped to be shape `(3, 2)`, and `coil_trans` is the values of `loc` rearranged a certain way to make a transformation matrix. Look more closely at the code in this PR, especially the `_loc_to_eeg_loc` and `_loc_to_coil_trans` functions and it should be clear.

**Comment by Eric89GXL** [comment]
In other words, I don't see any "differences" other than we convert from `loc` to `eeg_loc` or `coil_trans` (depending on MEG or EEG ch type) on-the-fly now, instead of storing the values in `info['chs'][ii][...]` once during read. Storing the values can be convenient for use, but it's dangerous in practice now that we are changing these values in `bti` and `maxwell`. People can mistakenly think that changing `info['chs'][0]['coil_trans']` will change what gets written to disk, and it doesn't. Only changing `info['chs'][0]['loc']` will. An alternative I considered is to write more sanity checking logic and throw errors if people change one value and not the other, but that seems less clean than avoiding errant changes in the first place by getting people to use conversion functions.

**Comment by agramfort** [comment]
ok I surrender again :)

**Comment by Eric89GXL** [comment]
That's better :)

**Comment by agramfort** [comment]
yes master :)

**Pull request merged**